starter_code/src/movies.js

```diff
@@ -1,19 +1,157 @@
 /* eslint no-restricted-globals: 'off' */
 
-// Iteration 1: All rates average - Get the average of all rates with 2 decimals 
+// Iteration 1: All rates average - Get the average of all rates with 2 decimals
+function ratesAverage(movies) {
+  let rate = [];
+  movies.forEach(function(element) {
+    rate.push(Number(element.rate));
+  });
 
- 
-// Iteration 2: Drama movies - Get the average of Drama Movies
+  const average =
+    rate.reduce(function(total, amount) {
+      return total + amount;
+    }, 0) / movies.length;
+  return parseFloat(average.toFixed(2));
+}
 
+// Iteration 2: Drama movies - Get the average of Drama Movies
+function dramaMoviesRate(dramas) {
+  const filtered = dramas.filter(function(movie) {
+    if (movie.genre.indexOf("Drama") === 0) {
+      return true;
+    }
+  });
+  const dramaRating = ratesAverage(filtered);
+  if (filtered.length === 0) {
+    return 0;
+  }
+  return dramaRating;
+}
 
 // Iteration 3: Ordering by duration - Order by time duration, ascending (in growing order)
 
+function orderByYear(movies) {
```

Comment:
function orderByYear(movies) {
  movies.sort(function(a, b) {
    if (a.year === b.year) {
      return a.title.localeCompare(b.title);
    } else {
      return a.year - b.year;
    }
  });
  return movies;
}

Here you could also use .localeCompare() in order to not have to write the two if statements.